server/server/src/account_service.rs

```diff
@@ -125,44 +127,71 @@ pub fn username_from_public_key(
         .unwrap_or(Err(ClientError(GetUsernameError::UserNotFound)))
 }
 
-pub async fn get_usage(
+pub async fn get_usage_info(
     context: RequestContext<'_, GetUsageRequest>,
 ) -> Result<GetUsageResponse, ServerError<GetUsageError>> {
-    let db = context.server_state.index_db.lock()?;
-    let cap = db
-        .accounts
-        .data()
-        .get(&Owner(context.public_key))
-        .ok_or(ClientError(GetUsageError::UserNotFound))?
-        .billing_info
-        .data_cap();
-
-    let usages = get_usage_helper(&db, &context.public_key)?;
-
-    Ok(GetUsageResponse { usages, cap })
+    let mut lock = context.server_state.index_db.lock()?;
+    let db = lock.deref_mut();
+
+    let cap = get_cap(db, &context.public_key)?;
+
+    let tree = ServerTree::new(
+        Owner(context.public_key),
+        &mut db.owned_files,
+        &mut db.shared_files,
+        &mut db.file_children,
+        &mut db.metas,
+    );
+
+    match tree {
+        Ok(t) => {
+            let usages = get_usage(&t.to_lazy(), db.sizes.data(), None)?;
+            Ok(GetUsageResponse { usages, cap })
+        }
+        Err(_) => Err(ClientError(GetUsageError::UserNotFound)),
+    }
```

Comment:
is there a better way to propagate this error? ideally it would be `ServerTree::new()?.to_lazy()` but that throws an error conversion exception.